smarts/core/plan.py

```diff
@@ -303,7 +303,7 @@ def create_route(self, mission: Mission) -> Mission:
         self._mission = mission or Mission.random_endless_mission(self._road_map)
 
         if not self._mission.has_fixed_route:
-            self._route = RoadMap.Route()
+            self._route = self._road_map.empty_route()
```

Comment:
Normally, I would say that the default should be static but I could understand why it might be useful to use a dynamic call here.

Reply:
Yeah, in general, _all_ creation of sub-classes of `RoadMap` should be done this way.  Or put another way, _only_ extensions of the `RoadMap` class should be responsible for creating their sub-classes.  For example, a `SumoRoadNetwork` extension should create `SumoRoadNetwork.Lane` objects, and objects of this type of map will expect such "lane" objects to be `SumoRoadNetwork.Lane` objects too, not merely `RoadMap.Lane` objects.  The same goes for `Road` and `Route` objects.

I got careless in the above, but thankfully caught myself when I added `distance_between()` to the `Route` sub-class.